src/main/java/com/radixdlt/crypto/hdwallet/BIP32Path.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.crypto.hdwallet;
+
+
+/**
+ * A wrapper around some underlying BIP32Path implementation, that is easily swappable.
+ * This class doesn't and shouldn't inherit from said wrapped implemetation, but rather use
+ * it as a trampoline. Since all interfaces are forwarded to the underlying wrapped implementation
+ * this class should really be trivial.
+ *
+ * However, users are discouraged to construct instances of this class directly, they should
+ * rather be using `DefaultHDPath`.
+ *
+ */
+public class BIP32Path implements HDPath {
```

Comment:
this class is not needed, but it does provide one level more of abstraction of `HDPath` implementation, i.e. `BitcoinJBIP32Path `.

So please note that we can delete this class, and rename `BitcoinJBIP32Path -> BIP32Path` instead.